metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -290,19 +306,21 @@ class BuildServerConnection private (
     }
 
   }
+
   private def register[T: ClassTag](
       action: MetalsBuildServer => CompletableFuture[T],
       onFail: => Option[(T, String)] = None,
+      isCompile: Boolean = false,
   ): CompletableFuture[T] = {
     val original = connection
     val actionFuture = original
       .flatMap { launcherConnection =>
         val resultFuture = action(launcherConnection.server)
-        ongoingRequests.add(
-          Cancelable(() =>
-            Try(resultFuture.completeExceptionally(new InterruptedException()))
-          )
-        )
+        val cancelable = Cancelable { () =>
+          Try(resultFuture.cancel(true))
+        }
+        if (isCompile) ongoingCompilations.add(cancelable)
+        else ongoingRequests.add(cancelable)
```

Comment:
`MutableCancelable.remove` is never used - futures are never removed from that queue which is a memory leak

Reply:
Good catch, I think this was an issue previously actually also. Added an explicit step afterwards to remove the cancelable.

We might need to fix it in a couple of places in the codebase, but I will follow up with another PR.